src/Controllers/ImageController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace JustBetter\GlideDirective\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Illuminate\Routing\Controller;
+use Illuminate\Support\Str;
+use JustBetter\GlideDirective\Responsive;
+use Statamic\Contracts\Assets\Asset;
+use Statamic\Contracts\Imaging\ImageManipulator;
+use Statamic\Facades\Asset as AssetFacade;
+use Statamic\Facades\Image;
+use Statamic\Imaging\GlideImageManipulator;
+
+class ImageController extends Controller
+{
+    public function getPreset(Request $request, string $preset, string $fit, string $signature, string $file, string $format): Response
+    {
+        $asset = AssetFacade::findByUrl(Str::start($file, '/'));
+
+        if (! $asset) {
+            abort(404);
+        }
+
+        /** @var GlideImageManipulator $manipulator */
+        $manipulator = self::getManipulator($asset, $preset, $fit, $format);
+        $path = $manipulator->build();
+
+        $publicPath = public_path($path);
+
+        if (! file_exists($publicPath)) {
+            abort(404);
+        }
+
+        /* @phpstan-ignore-next-line */
+        $contentType = $asset->mimeType();
+        $fileContent = file_get_contents($publicPath) ?: '';
+
+        return response($fileContent)
+            ->header('Content-Type', $contentType)
+            ->header('Cache-Control', 'public, max-age=31536000');
```

Comment:
We could respond with https://symfony.com/doc/current/components/http_foundation.html#:~:text=Alternatively%2C%20if%20you,2%0A3%0A4

```suggestion
        return new BinaryFileResponse($publicPath)
```

Which is better suited for large binary files as it streams the image.